Finished the 404 page

teddy-f-47 opened the pull request:
What's changed:
- ui/src/assets/svg/404.png : Replaced the png file with a new one that has a higher resolution.
- ui/src/views/PageNotFoundView.vue : Added the 404 image and text, finished styling the page with an alternative image size (and button size) for mobile device.

To do:
- ui/src/views/PageNotFoundView.vue : Redirect after 3 seconds based on user type. However, in my opinion, the "back to home" button alone is sufficient, so the redirection will only be added later if we have enough time.

<img width="480" alt="screenshotPage404" src="https://user-images.githubusercontent.com/64476430/170820113-030ef420-3bd0-4cf6-a457-3ee56322455e.png">

<img width="480" alt="screenshotPage404-mobile" src="https://user-images.githubusercontent.com/64476430/170821928-9902cea3-c5c6-41ec-add4-14f1e131956a.png">

teddy-f-47 committed:
Finished the 404 page

What's changed:
- ui/src/assets/svg/404.png : Replaced the png file with a new one with a higher resolution.
- ui/src/views/PageNotFoundView.vue : Added the 404 image and text, finished styling the page.

To do:
- ui/src/views/PageNotFoundView.vue : Redirect after 3 seconds based on user type. However, in my opinion, the "back to home" button alone is sufficient, so the redirection will only be added later if we have enough time.

artikandri commented on ui/src/views/PageNotFoundView.vue:
```diff
@@ -21,11 +28,21 @@
 
 <script>
-  export default { name: "ErrorView" };
+  export default { name: "ErrorView",
```
Can you replace the name to PageNotFoundView.vue?

teddy-f-47 replied:
Okay, so I will change `name: "ErrorView"` to `name: "PageNotFoundView.vue"`, right?

artikandri replied:
To PageNotFoundView, without the .vue extension. Ideally component name should be in pascal case, more than one word (tho I sometimes break this pattern), and follow the component file name.

artikandri commented on ui/src/views/PageNotFoundView.vue:
```diff
@@ -21,11 +28,21 @@
+    variables() {
+      return require("@styles/variables.js");
+    },
+    mainPicture() {
```
Is it possible to replace the pić with svg? I think I have exported the svg but Im not sure

teddy-f-47 replied:
The issue with the SVG file is that it has the text "designed by x" which does not look nice. That's why I converted the SVG file into PNG and removed that text. I could not find a good SVG editor to do that correctly.

artikandri replied:
Aha, I see. Alright, lets leave that as png for now.. I think it can be done with vector editor such as Corel or ai but it takes time, better to use png then

artikandri commented on ui/src/views/PageNotFoundView.vue:
```diff
@@ -6,12 +6,19 @@
           large
           to="/"
+          :color="variables.default.primary"
+          :width="imageWidth"
```
Is there any reason on why the width props is bound to the data? Cant the value be string  (width="100") for example

Also the var name is imageWidth while the element is button

artikandri replied:
Also, have you checked the mobile view? (Use mobile dev tools in dev tools, there is a phone icon near the top right screen, click on it and you can choose the device).

teddy-f-47 replied:
I will put all those elements in a `col` and just set the width for that `col`. Also yes I did check the mobile view on various devices available in the dev tools.

artikandri replied:
Great, thank you! I will initialize the variables as global variables, as for now please wait or probably work on some other things/cards..

artikandri commented on ui/src/views/PageNotFoundView.vue:
```diff
@@ -6,12 +6,19 @@
+        <p>The page either doesn't exist or you don't have access to this page</p>
+        <center>
+          <v-img :src="mainPicture" :width="imageWidth" contain />
+        </center>
```
I think width doesnt have to be bound to the data here, instead we can use string. And for me personally,  I think its better to use max-height attribute rather than width. Please check the vuetify doc on v-img for more info

artikandri commented:
Thanks for the quick work! I have pointed out some things, can you check when you have time?

teddy-f-47 committed:
Update PageNotFoundView.vue

- ui/src/views/PageNotFoundView.vue : Changed imageWidth unit from percent to vw in order to make better mobile appearance.

teddy-f-47 committed:
Update PageNotFoundView.vue

ui/src/views/PageNotFoundView.vue : Added alternative size for mobile device

artikandri commented on ui/src/views/PageNotFoundView.vue:
```diff
@@ -41,6 +55,16 @@
       return require("@assets/svg/404.png");
     }
+  },
+  methods: {
```
One way is to do this, the other way is to use CSS and target the media query instead. But this is fine, I will make a global mixin so that we dont have to specify this in every file.
